javascript/javascript3/week2/promise-visual/main.js

```diff
@@ -0,0 +1,125 @@
+// modify this page only
+
+// This code will move the li to the position 100, 100. Calling moveElement will return a promise that resolves after the li element has been moved.
+const li1 = document.querySelector('li');
+const li2 = document.querySelector('ul.marks li:nth-child(2)');
+const li3 = document.querySelector('ul.marks li:nth-child(3)');
+
+function displayThumbsUp() {
+    const thumbsUp = document.querySelector('.thumbs-up');
+    const div = document.createElement('div');
+    div.innerHTML = thumbsUp.textContent;
+    div.style.marginLeft = '200px';
+    div.style.marginTop = '150px';
+    div.style.fontSize = '80px';
+    document.body.appendChild(div);
+}
+
+//using promise
+function translateOneByOne() {
+    moveElement(li1, { x: 20, y: 300 })
+        .then(() => {
+            console.log('Red element has been moved');
+        })
+        .then(() => {
+            moveElement(li2, { x: 400, y: 300 })
+                .then(() => {
+                    console.log('Blue element has been moved');
+                })
+                .then(() => {
+                    moveElement(li3, { x: 400, y: 20 })
+                        .then(() => {
+                            console.log('Green element has been moved');
+                        })
+                        .then(() => {
+                            displayThumbsUp();
+                        });
+                });
+        });
+} //end of function
+
+//translateOneByOne();
+
+function translateAllAtOnce() {
+    const promise1 = moveElement(li1, { x: 20, y: 300 }).then(() => {
+        console.log('first promise');
+    });
+    const promise2 = moveElement(li2, { x: 400, y: 300 }).then(() => {
+        console.log('second promise');
+    });
+    const promise3 = moveElement(li3, { x: 400, y: 20 }).then(() => {
+        console.log('third promise');
+    });
+
+    Promise.all([promise1, promise2, promise3]).then((values) => {
```

Comment:
Promise.all() returns a prromise that can be awaited if thats what you mean

Reply:
Promise.all waits for all fulfillments (or the first rejection).  Isn't that true? otherwise what you suggest me to do?

Reply:
You write you are unable to put thumbsup. 
I'm thinking its a matter of putting your  displayThumbsUp(); into the callback

```
 Promise.all([promise1, promise2, promise3]).then((values) => { 
                 displayThumbsUp();
 })
```